datafusion/core/src/datasource/physical_plan/csv.rs

```diff
@@ -101,6 +218,39 @@ impl CsvExec {
             comment,
         }
     }
+}
+
+impl CsvExec {
+    /// Create a new CSV reader execution plan provided base and specific configurations
+    #[deprecated(since = "41.0.0", note = "use `CsvExec::builder` or `CsvExecBuilder`")]
```

Comment:
💯  for keeping (and deprecating the old API) to make migration / upgrade easier.